lib/versioning/poetry/transform.ts

```diff
@@ -0,0 +1,133 @@
+import { parse } from 'semver';
+import { VERSION_PATTERN, RANGE_COMPARATOR_PATTERN } from './patterns';
+
+function parseLetterTag(letter?: string, number?: string) {
+  if (letter != undefined) {
+    if (number == undefined) {
+      number = '0';
+    }
+    // apply the same normalizations as poetry
+    var spellings = {
+      alpha: 'a',
+      beta: 'b',
+      c: 'rc',
+      pre: 'rc',
+      preview: 'rc',
+      r: 'post',
+      rev: 'post',
+    };
+    return { letter: spellings[letter] || letter, number: number };
+  }
+  if (letter == undefined && number != undefined) {
+    return { letter: 'post', number: number };
+  }
+}
+
+function notEmpty(s: string): boolean {
+  return s !== '';
+}
+
+// Parse versions like poetry.core.masonry.version.Version does (union of SemVer
+// and PEP440, with normalization of certain prerelease tags), and emit in SemVer
+// format. NOTE: this silently discards the epoch field in PEP440 versions, as
+// it has no equivalent in SemVer.
```

Comment:
```suggestion
/**
 * Parse versions like poetry.core.masonry.version.Version does (union of SemVer
 * and PEP440, with normalization of certain prerelease tags), and emit in SemVer
 * format. NOTE: this silently discards the epoch field in PEP440 versions, as
 * it has no equivalent in SemVer.
 */
```

We should use jsdoc style, please update others too.

Reply:
Done.